tools/analysis/programs.go

```diff
@@ -145,12 +172,18 @@ func (programs Programs) check(
 						return nil, err
 					}
 
-					elaboration = programs[importedLocation].Checker.Elaboration
+					// If the imported program has a checker, use its elaboration for the import
+					if programs[importedLocation].Checker != nil {
+						elaboration = programs[importedLocation].Checker.Elaboration
+					}
+
+					// If the imported program had an error while loading, record it
+					loadError = programs[importedLocation].LoadError
 				}
 
-				return sema.ElaborationImport{
+				return &sema.ElaborationImport{
 					Elaboration: elaboration,
-				}, nil
+				}, loadError
```

Comment:
There shouldn't be any need to return a pointer here.

Also, the common pattern is to either return a value, or an error, not both.

```suggestion
				}
				if loadError != nil {
					return nil, loadError
				}

				return sema.ElaborationImport{
					Elaboration: elaboration,
				}, nil
```